File: src/main/kotlin/cn/eusunpower/config/BeanConfig.java
package cn.eusunpower.config;

import org.hibernate.SessionFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.annotation.Resource;
import javax.persistence.EntityManagerFactory;


@Configuration
public class BeanConfig {

    @Resource
    private EntityManagerFactory entityManagerFactory;

    @Bean
    public SessionFactory getSessionFactory() {
        if (entityManagerFactory.unwrap(SessionFactory.class) == null) {
            throw new NullPointerException("factory is not a hibernate factory");
        }
        return entityManagerFactory.unwrap(SessionFactory.class);
    }
}

File: src/main/kotlin/cn/eusunpower/config/BeanConfig.kt
package cn.eusunpower.config

import org.hibernate.SessionFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import javax.annotation.Resource
import javax.persistence.EntityManagerFactory


@Configuration
class BeanConfig {

    @Resource
    lateinit var entityManagerFactory: EntityManagerFactory

    @Bean
    fun sessionFactory(): SessionFactory {
        return entityManagerFactory.unwrap(SessionFactory::class.java)
    }

    @Bean
    fun passwordEncoder(): PasswordEncoder = BCryptPasswordEncoder()
}
